src/api/Controllers/UploadController.cs

```diff
@@ -248,13 +250,43 @@ public async Task<ActionResult<int>> UploadFileAsync(int workgroupId, IFormFile
         }
     }
 
+    internal List<int> ParseIds(string ids)
+    {
+        if (string.IsNullOrEmpty(ids))
+        {
+            return new List<int>();
+        }
+
+        return ids.Split(',')
+            .Select(s =>
+            {
+                if (!int.TryParse(s, out var num))
+                {
+                    logger.LogError($"Invalid code list value: {s}");
+                    throw new FormatException($"Invalid code list value: {s}");
+                }
+
+                return num;
+            }).ToList();
+    }
+
+    internal List<Codelist> GetCodelists(ICollection<int> codelistIds)
+    {
+        return context.Codelists.Where(c => codelistIds.Contains(c.Id)).ToList();
+    }
+
+    internal List<T> CreateLayerCodes<T>(List<Codelist> codelists)
+        where T : ILayerCode, new()
+    {
+        return codelists.Select(c => new T { Codelist = c, CodelistId = c.Id }).ToList();
+    }
+
     internal List<int> ParseMultiValueCodeListIds(LithologyImport lithologyImport)
     {
         // Select all code list ids of all multi value code list properties.
         var splittedList = new[] { lithologyImport.ColorIds, lithologyImport.OrganicComponentIds, lithologyImport.GrainShapeIds, lithologyImport.GrainGranularityIds, lithologyImport.Uscs3Ids, lithologyImport.DebrisIds }
             .SelectMany(str => str.Split(','))
             .ToList();
-
         return splittedList.Where(s => !string.IsNullOrEmpty(s)).Select(int.Parse).ToList() ?? new List<int>();
```

Comment:
PP: Das ganze liesse sich in einem Statement zusammenfassen und gleich zurückgeben.